packages/contracts/test/voting/multisig.ts

```diff
@@ -90,119 +95,138 @@ describe('Multisig', function () {
     return addresses;
   }
 
-  describe('initialize: ', async () => {
+  describe('initialize:', async () => {
     it('reverts if trying to re-initialize', async () => {
-      await multisig.initialize(dao.address, addresslist(5), {
-        onlyListed,
-        minApprovals,
-      });
+      await multisig.initialize(dao.address, addresslist(5), multisigSettings);
 
       await expect(
-        multisig.initialize(dao.address, addresslist(5), {
-          onlyListed,
-          minApprovals,
-        })
+        multisig.initialize(dao.address, addresslist(5), multisigSettings)
       ).to.be.revertedWith(ERRORS.ALREADY_INITIALIZED);
     });
+
+    it('adds the initial addresses to the address list', async () => {
+      expect(await multisig.addresslistLength()).to.equal(0);
+
+      multisigSettings.minApprovals = 2;
+      await multisig.initialize(dao.address, addresslist(2), multisigSettings);
+
+      expect(await multisig.addresslistLength()).to.equal(2);
+      expect(await multisig.isListed(signers[0].address)).to.equal(true);
+      expect(await multisig.isListed(signers[1].address)).to.equal(true);
+    });
+
+    it('should set the `minApprovals`', async () => {
+      await multisig.initialize(dao.address, addresslist(5), multisigSettings);
+      expect((await multisig.multisigSettings()).minApprovals).to.be.eq(
+        multisigSettings.minApprovals
+      );
+    });
+
+    it('should set `onlyListed`', async () => {
+      await multisig.initialize(dao.address, addresslist(5), multisigSettings);
+      expect((await multisig.multisigSettings()).onlyListed).to.be.eq(
+        multisigSettings.onlyListed
+      );
+    });
+
+    it('should emit `MultisigSettingsUpdated` during initialization', async () => {
+      await expect(
+        multisig.initialize(dao.address, addresslist(5), multisigSettings)
+      )
+        .to.emit(multisig, MULTISIG_EVENTS.MULTISIG_SETTINGS_UPDATED)
+        .withArgs(multisigSettings.onlyListed, multisigSettings.minApprovals);
+    });
+  });
+
+  describe('updateMultisigSettings:', async () => {
+    beforeEach(async () => {
+      await multisig.initialize(dao.address, addresslist(5), multisigSettings);
+    });
+
+    it('should not allow to set minApprovals larger than the address list length', async () => {
+      let addresslistLength = (await multisig.addresslistLength()).toNumber();
+
+      multisigSettings.minApprovals = addresslistLength + 1;
+
+      await expect(
+        multisig.updateMultisigSettings(multisigSettings)
+      ).to.be.revertedWith(
+        customError(
+          'MinApprovalsOutOfBounds',
+          addresslistLength,
+          multisigSettings.minApprovals
+        )
+      );
+    });
+
+    it('should not allow to set minApprovals would be set to 0', async () => {
```

Comment:
grammar error.

Reply:
Fixed in 57174b98c910188a41e81ac5457f041910a75a0b